docs/nlu/choosing-a-pipeline.rst

```diff
@@ -15,76 +13,133 @@ it on your dataset.
 
-You can read more about this topic `here <https://medium.com/rasa-blog/supervised-word-vectors-from-scratch-in-rasa-nlu-6daf794efcd8>`__ .
+    1. Tokenizaion
+    2. Featuirzation
+    3. Entity Recognition / Intent Classification / Response Selectors
 
-MITIE
-~~~~~
+Tokenization
+~~~~~~~~~~~~
+If your chosen language is whitespace-tokenized (words are separated by spaces), you
+can use the ``WhitespaceTokenizer``. If this is not the case you should use a different tokenizer.
+We support a number of different :ref:`tokenizers <tokenizers>`, or you can :ref:`create your own <custom-nlu-components>`.
 
-You can also use MITIE as a source of word vectors in your pipeline, see :ref:`section_mitie_pipeline`. The MITIE backend performs well for small datasets, but training can take very long if you have more than a couple of hundred examples.
+.. note::
+    Some components further down the pipeline may require a specific tokenizer. You can find those requirements
+    on the individual components in :ref:`components`. If a required component is missing inside the pipeline, an
+    error will be thrown.
+
+Featurization
+~~~~~~~~~~~~~
+You need to decide whether to use components that provide pre-trained word embeddings or not.
+
+If you do not use any pre-trained word embeddings, your word vectors will be customised for your domain. For example,
+in general English, the word "balance" is closely related to "symmetry", but very different to the word "cash". In a
+banking domain, "balance" and "cash" are closely related and you'd like your model to capture that. If you don't
+use any pre-trained word embeddings inside your pipeline, you are not bound to a specific language and domain.
+Thus, you should only use featurizers from the category `sparse` featuirzers, such as
+``CountVectorsFeaturizer`` or ``RegexFeaturizer``.
+
+The advantage of using pre-trained word embeddings in your pipeline is that if you have a training example like:
+"I want to buy apples", and Rasa is asked to predict the intent for "get pears", your model already knows that the
+words "apples" and "pears" are very similar. This is especially useful if you don't have large enough training data.
+We support a few components that provide pre-trained word embeddings:
+
+1. ``MitieFeaturizer``
+2. ``SpacyFeaturizer``
+3. ``ConveRTFeaturizer``
+4. ``LanguageModelFeaturizer``
+
+The advantage of the ``ConveRTFeaturizer`` is that it doesn't treat each word of the user message independently, but
+creates a contextual vector representation for the complete sentence. For example, if you
+have a training example, like: "can I book a car?", and Rasa is asked to predict the intent for "I need a ride from
+my place", since the contextual vector representation for both examples are already very similar, the intent classified
+for both is highly likely to be the same. This is also useful if you don't have large enough training data.
+
+TODO when to use what featurizer
```

Comment:
@dakshvar22 @Ghostvv When do we recommend users to use what kind of dense featurizer? General motivation is clear, but do you have good arguments when to use one specific language model featurizer rather than another one?

Reply:
if language is English, then convert

Reply:
updated the documentation, can you please have a look at this part of the docs? thanks.

Reply:
Since we describe convert here, I think, it makes sense to also describe spacy, mite and lm briefly